src/components/DownloadMailingListButton.tsx

```diff
@@ -22,30 +25,54 @@ export function DownloadMailingListButton (props: DownloadButtonProps) {
-
-        if (ownersArray) {
-            ownersArray.forEach(function (arrayItem) {
-                doNotMail = arrayItem.Do_Not_Mail
-                returnToSender = arrayItem.Return_to_Sender
-                let ownerNameOnTitle = arrayItem.Contact_Type === 'Owner' ? arrayItem?.Company : ownerData?.Name
-                postalAddress = arrayItem.Postal_Address ? arrayItem.Postal_Address.split(', ')[0]  :  `${arrayItem.Postal_Unit ? `${arrayItem.Postal_Unit}/` : ''} ${arrayItem.Postal_Street_No} ${arrayItem.Postal_Street}`
-                const isDupe = ownerContactDupeRemoval.includes(`${postalAddress}-${arrayItem?.Name}`)
+        let nameOnTitleContact_list = propertyObject.NameOnTitleContactDetails
+        let contactArrayStr =  `${propertyObject.Contacts_Cache}`
+        
+       
+        let contactArray = JSON.parse(contactArrayStr)
+        if (contactArray == null){
+            contactArray = []
+        }
+        if (contactArray.length > 1) {
+            for (let i = 0; i < contactArray.length; ++i)
+                for (let j = 0; j < contactArray.length; ++j)
+                    if (i !== j && contactArray[i].id === contactArray[j].id)
+                    contactArray.splice(j, 1); 
+            contactArray = contactArray
+        }
+        
+        if (contactArray.toString()) {
+            for (let i = 0; i < contactArray.length; ++i)
+            {
+                let name_on_title = nameOnTitleContact_list?.find((owner) => owner.Contact.id === contactArray[i].id)
+                doNotMail = contactArray[i].Do_Not_Mail
+                returnToSender = contactArray[i].Return_to_Sender
+                postalAddress = contactArray[i].Postal_Address ? contactArray[i].Postal_Address.split(', ')[0]  :  `${contactArray[i].Postal_Unit ? `${contactArray[i].Postal_Unit}/` : ''} ${contactArray[i].Postal_Street_No} ${contactArray[i].Postal_Street}`
+                const isDupe = ownerContactDupeRemoval.includes(`${postalAddress}-${contactArray[i]?.Name}`)
+                
                 if (!doNotMail && !returnToSender) {
-                    if (!postalAddress.includes('null') && !isExactMatchForSearchAddress) {
+                    if (!postalAddress.includes('null') && !isExactMatchForSearchAddress){
                         if (!isDupe) {
-                            const lastMailed = arrayItem.Last_Mailed || 'Last mailed has not been found'
-                            ownerContactDupeRemoval.push(`${postalAddress}-${arrayItem?.Name}`)
-                            csvRowsForProperty += `"${propertyAddress}","${ownerNameOnTitle || ''}","${arrayItem?.First_Name} ${arrayItem?.Last_Name}","${postalAddress}","${arrayItem?.Postal_Suburb}","${arrayItem?.Postal_State}","${arrayItem?.Postal_Postcode}","${arrayItem?.Salutation_Dear}","${arrayItem?.Email}",${propertyType},${lastMailed}\r\n`
+                            const lastMailed = contactArray[i].Last_Mailed || 'Last mailed has not been found'
```

Comment:
instead of referencing `contactArray[i]` all the time, declare a variable `currentContact = contactArray[i]`